src/components/ui/Select/Select.tsx

```diff
@@ -40,10 +43,10 @@ const Select = ({
 
       <select
         id={name}
-        name={name}
         className="govuk-select lbh-select"
+        data-testid="select"
         {...rest}
-        {...register(name)}
+        {...(register && register(name))}
```

Comment:
Let's use optional chaining instead
```suggestion
        {...register?.(name)}
```